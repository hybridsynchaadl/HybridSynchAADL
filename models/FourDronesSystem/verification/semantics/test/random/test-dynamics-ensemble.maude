set show timing off .

load ../../interpreter-random.maude
load ../syntax/module-thermostat-noparam.maude 


-------------------------------------------------------
mod TEST-ENSEMBLE-DYNAMICS is
  including THERMOSTAT-SYSTEM-NOPARAM .
  including MODEL-INIT .
  including THREAD-DYNAMICS .
  including ENV-DYNAMICS .
  including ENSEMBLE-DYNAMICS .
  including TRANSFER-INPUT .
  including TRANSFER-RESULTS .
  including CONC-REAL-EXP .

  op ensemble : -> Object .
  eq ensemble = initialize(model) .
endm

rew execSample({ensemble, 0}) .
rew execResponse({ensemble, 0}) .

rew execSampleSub({ensemble, 0}) .
rew transferInputs(execSampleSub({ensemble, 0})) .
rew propagateExecSub(transferInputs(execSampleSub({ensemble, 0}))) .
rew transferResults(propagateExecSub(transferInputs(execSampleSub({ensemble, 0})))) .
rew execResponseSub(transferResults(propagateExecSub(transferInputs(execSampleSub({ensemble, 0}))))) .


rew executeStep({ensemble, 0}) .
