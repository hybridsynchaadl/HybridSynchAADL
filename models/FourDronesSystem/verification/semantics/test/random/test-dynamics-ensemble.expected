==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : execSample({ensemble,0}) .
rewrites: 113
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [0.0] # [false] >),
    subcomponents : (< lower : Data | properties : DataModel::InitialValue => {
    {10}},features : none,subcomponents : none,connections : empty,value : [
    1.0e+1] # [true] > < upper : Data | properties : DataModel::InitialValue =>
    {{30}},features : none,subcomponents : none,connections : empty,value : [
    3.0e+1] # [true] > < curr : Data | properties : DataModel::InitialValue =>
    {{0}},features : none,subcomponents : none,connections : empty,value : [
    0.0] # [true] >),connections : empty,varGen : < "top.controller" >,
    currState : l1,completeStates : l1,variables : x : Real,transitions : ((l1
    -[on dispatch]-> l2 {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower])(on !) (elsif(v[x] > c[upper])off !) end if ; c{curr} := v[x]}) ; l2
    -[otherwise]-> l1 {skip}),loopBound : 0,transBound : 2 > < environment :
    EnvWithTimer | properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : controller > < off : EnvInPort | properties : none,
    content : * # [false],envCache : null(Unit),target : controller > < out :
    EnvOutPort | properties : none,content : [0.0] # [false],envCache : null(
    Real),target : controller >),subcomponents : < x : Data | properties :
    DataModel::InitialValue => {{16}},features : none,subcomponents : none,
    connections : empty,value : [1.6e+1] # [true] >,connections : x ==> out,
    varGen : < "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1
    -[off]-> l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x]
    + (v[t] * [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,0}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : execResponse({ensemble,0}) .
rewrites: 113
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [0.0] # [false] >),
    subcomponents : (< lower : Data | properties : DataModel::InitialValue => {
    {10}},features : none,subcomponents : none,connections : empty,value : [
    1.0e+1] # [true] > < upper : Data | properties : DataModel::InitialValue =>
    {{30}},features : none,subcomponents : none,connections : empty,value : [
    3.0e+1] # [true] > < curr : Data | properties : DataModel::InitialValue =>
    {{0}},features : none,subcomponents : none,connections : empty,value : [
    0.0] # [true] >),connections : empty,varGen : < "top.controller" >,
    currState : l1,completeStates : l1,variables : x : Real,transitions : ((l1
    -[on dispatch]-> l2 {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower])(on !) (elsif(v[x] > c[upper])off !) end if ; c{curr} := v[x]}) ; l2
    -[otherwise]-> l1 {skip}),loopBound : 0,transBound : 2 > < environment :
    EnvWithTimer | properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : controller > < off : EnvInPort | properties : none,
    content : * # [false],envCache : null(Unit),target : controller > < out :
    EnvOutPort | properties : none,content : [0.0] # [false],envCache : null(
    Real),target : controller >),subcomponents : < x : Data | properties :
    DataModel::InitialValue => {{16}},features : none,subcomponents : none,
    connections : empty,value : [1.6e+1] # [true] >,connections : x ==> out,
    varGen : < "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1
    -[off]-> l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x]
    + (v[t] * [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,0}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : execSampleSub({ensemble,0}) .
rewrites: 253
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [0.0] # [false] >),
    subcomponents : (< lower : Data | properties : DataModel::InitialValue => {
    {10}},features : none,subcomponents : none,connections : empty,value : [
    1.0e+1] # [true] > < upper : Data | properties : DataModel::InitialValue =>
    {{30}},features : none,subcomponents : none,connections : empty,value : [
    3.0e+1] # [true] > < curr : Data | properties : DataModel::InitialValue =>
    {{0}},features : none,subcomponents : none,connections : empty,value : [
    0.0] # [true] >),connections : empty,varGen : < "top.controller" >,
    currState : l1,completeStates : l1,variables : x : Real,transitions : ((l1
    -[on dispatch]-> l2 {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower])(on !) (elsif(v[x] > c[upper])off !) end if ; c{curr} := v[x]}) ; l2
    -[otherwise]-> l1 {skip}),loopBound : 0,transBound : 2 > < environment :
    EnvWithTimer | properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< on : EnvInPort | properties : none,content : * # [false],envCache :
    null(Unit),target : controller > < off : EnvInPort | properties : none,
    content : * # [false],envCache : null(Unit),target : controller > < out :
    EnvOutPort | properties : none,content : [1.6260851631271851e+1] # [true],
    envCache : null(Real),target : controller >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [1.6e+1] # [true] >,
    connections : x ==> out,varGen : < "top.environment" >,currMode : l1,jumps
    : ((l1 -[on]-> l1) ; (l1 -[off]-> l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),
    flows : ((l1 [x(t)= (c[x] + (v[t] * [[2.0000000000000001e-1]]))]) ; l2 [x(
    t)= (c[x] - (v[t] * [[1.0000000000000001e-1]]))]),sampling : controller :(
    0,2),response : controller :(3,4),timer :
    6727472470607199929743149/5158083495821457911447552,samplingIDs :
    controller,responseIDs : controller >),connections : (((controller .. on)
    --> (environment .. on)) ; ((controller .. off) --> (environment .. off)) ;
    (environment .. out) --> (controller .. temp)) >,2}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : transferInputs(execSampleSub({ensemble,0}))
    .
rewrites: 283
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [1.6260851631271851e+1] # [true],cache : [0.0]
    # [false] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [0.0] # [true] >),connections : empty,varGen :
    < "top.controller" >,currState : l1,completeStates : l1,variables : x :
    Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[curr] + f[temp])
    / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[upper])off !) end if
    ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),loopBound : 0,transBound
    : 2 > < environment : EnvWithTimer | properties : ((
    TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation =>
    {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.6e+1] # [true] >,connections : x ==> out,varGen : < "top.environment" >,
    currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]-> l2) ; (l2 -[on]-> l1)
    ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer :
    6727472470607199929743149/5158083495821457911447552,samplingIDs :
    controller,responseIDs : controller >),connections : (((controller .. on)
    --> (environment .. on)) ; ((controller .. off) --> (environment .. off)) ;
    (environment .. out) --> (controller .. temp)) >,2}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : propagateExecSub(transferInputs(
    execSampleSub({ensemble,0}))) .
rewrites: 492
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [true] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [1.6260851631271851e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [8.1304258156359257] # [true] >),connections :
    empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.6e+1] # [true] >,connections : x ==> out,varGen : < "top.environment" >,
    currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]-> l2) ; (l2 -[on]-> l1)
    ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer :
    6727472470607199929743149/5158083495821457911447552,samplingIDs :
    controller,responseIDs : controller >),connections : (((controller .. on)
    --> (environment .. on)) ; ((controller .. off) --> (environment .. off)) ;
    (environment .. out) --> (controller .. temp)) >,4}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : transferResults(propagateExecSub(
    transferInputs(execSampleSub({ensemble,0})))) .
rewrites: 516
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [1.6260851631271851e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [8.1304258156359257] # [true] >),connections :
    empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [true],envCache : null(Unit),target : controller > < off
    : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.6e+1] # [true] >,connections : x ==> out,varGen : < "top.environment" >,
    currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]-> l2) ; (l2 -[on]-> l1)
    ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer :
    6727472470607199929743149/5158083495821457911447552,samplingIDs :
    controller,responseIDs : controller >),connections : (((controller .. on)
    --> (environment .. on)) ; ((controller .. off) --> (environment .. off)) ;
    (environment .. out) --> (controller .. temp)) >,4}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : execResponseSub(transferResults(
    propagateExecSub(transferInputs(execSampleSub({ensemble,0}))))) .
rewrites: 729
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [1.6260851631271851e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [8.1304258156359257] # [true] >),connections :
    empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.6739148368728152e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,6}
==========================================
rewrite in TEST-ENSEMBLE-DYNAMICS : executeStep({ensemble,0}) .
rewrites: 730
result ConfigWithRand: {< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [1.6260851631271851e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [8.1304258156359257] # [true] >),connections :
    empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.6739148368728152e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,6}
Bye.
